ddl/ddl_api.go

```diff
@@ -551,6 +555,24 @@ func checkPriKeyConstraint(col *table.Column, hasDefaultValue, hasNullFlag bool,
 	return nil
 }
 
+func checkColumnValueConstraint(col *table.Column) error {
+	if col.Tp == mysql.TypeEnum || col.Tp == mysql.TypeSet {
+		valueMap := make(map[string]string, len(col.Elems))
+		for i := range col.Elems {
+			val := strings.ToLower(col.Elems[i])
+			if _, ok := valueMap[val]; ok {
+				tpStr := "ENUM"
+				if col.Tp == mysql.TypeSet {
+					tpStr = "SET"
+				}
+				return types.ErrDuplicatedValueInType.GenWithStackByArgs(col.Name, valueMap[val], tpStr)
```

Comment:
why not use col.Elems[i] to replace valueMap[val] here?

Reply:
In MySQL
```
mysql> create table t1 (a enum("B","b"));
ERROR 1291 (HY000): Column 'a' has duplicated value 'B' in ENUM
```
if  use col.Elems[i] to replace valueMap[val] here, the error msg return will be:
 ```
ERROR 1291 (HY000): Column 'a' has duplicated value 'b' in ENUM
```